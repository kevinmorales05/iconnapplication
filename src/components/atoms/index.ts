export * from './AccountItem';
export * from './ActionButton';
export * from './AddressItems';
export * from './AnnounceItem';
export * from './Avatar';
export * from './BackButton';
export * from './CardAction';
export * from './Checkbox';
export * from './ConnectionItem';
export * from './Container';
export * from './CustomModal';
export * from './CustomText';
export * from './DatePicker';
export * from './EmptyCardsCard';
export * from './EmptyQRCard';
export * from './InfoCard';
export * from './Input';
export * from './PaymentCard';
export * from './SafeArea';
export * from './Section';
export * from './Select';
export * from './ServiceCard';
export * from './Touchable';
