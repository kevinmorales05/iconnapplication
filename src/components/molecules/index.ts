export * from './Accordion';
export * from './AddressCard';
export * from './BasketCounter';
export * from './Button';
export * from './CardBilling';
export * from './CardDivided';
export * from './CardHorizontal';
export * from './CardImage';
export * from './CheckButton';
export * from './Code';
export * from './CountryCodeSelect';
export * from './EcommerceHeader';
export * from './FavoriteButton';
export * from './ImagesCarusel';
export * from './ImageZoom';
export * from './ListSwipeableItem';
export * from './ModalCard';
export * from './NavigationMenuItem';
export * from './OrderCard';
export * from './PriceWithDiscount';
export * from './QuantityProduct';
export * from './RadioButton';
export * from './Rating';
export * from './SearchBar';
export * from './SearchItem';
export * from './TaxInfoCard';
export * from './TextContainer';
export * from './TouchableText';
