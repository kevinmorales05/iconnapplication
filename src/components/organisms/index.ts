export * from './AboutEmail';
export * from './Alert';
export * from './Loading';
export * from './OtherInputMethods';
export * from './InvoicingHelper';
export * from './ResendInvoice';
export * from './AddressModalSelection';
export * from './ShippingDropdown';
