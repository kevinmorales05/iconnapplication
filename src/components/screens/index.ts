export * from './Billing';
export * from './InvoiceGenerated';
