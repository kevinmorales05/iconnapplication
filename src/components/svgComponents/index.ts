export * from './SearchLoupeDeleteSvg';
export * from './StarSvg';
export * from './HeartSvg';
export * from './HeartSvgOutline';
export * from './BasketSvg';
export * from './CloseSvg';
