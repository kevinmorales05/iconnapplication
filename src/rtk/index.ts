export * from './types';
export * from './store';
export * from './slices';
export * from './thunks';
