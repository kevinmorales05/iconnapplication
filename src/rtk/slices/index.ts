export * from './appSlice';
export * from './authSlice';
export * from './guestSlice';
