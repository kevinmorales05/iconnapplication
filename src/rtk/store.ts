import { configureStore } from '@reduxjs/toolkit';
import authReducer from './slices/authSlice';
import appReducer from './slices/appSlice';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

export const store = configureStore({
  reducer: {
    auth: authReducer,
    app: appReducer
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
