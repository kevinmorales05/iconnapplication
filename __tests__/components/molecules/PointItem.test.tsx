import React from 'react';
import { PointItem } from 'components';
import { PointInterface } from 'rtk';
import renderer from 'react-test-renderer';

const onPressNothing = () => {};

jest.mock('react-native-device-info', () => {
  return {
    getVersion: () => 4,
    hasNotch: () => true,
    getModel: () => 'iPhone 8 Plus'
  };
});
jest.mock('@react-native-google-signin/google-signin', () => {});
jest.mock('@react-native-async-storage/async-storage', () => {});
jest.mock('react-native-permissions', () => {});
jest.mock('@react-native-community/netinfo', () => {});
jest.mock('react-native-blob-util', () => {});
jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter');
jest.mock('redux-persist', () => {
  const real = jest.requireActual('redux-persist');
  return {
    ...real,
    persistReducer: jest.fn().mockImplementation((config, reducers) => reducers)
  };
});

describe('Testing on <PointItem />', () => {
  test('should match with the snapshot', () => {
    const snap = renderer
      .create(
        <PointItem
          onPress={onPressNothing}
          point={
            {
              address: 'string',
              googleMapsLink: 'string',
              id: 123,
              info: {},
              isActive: true,
              isUpdated: true,
              kmDistance: 'string',
              latitude: 345,
              longitude: 345,
              mallNumber: 1234,
              postalCode: 1234,
              shopName: 'string',
              shopNumber: 123,
              type: '7eleven'
            } as PointInterface
          }
          pointFound={false}
        />
      )
      .toJSON();

    expect(snap).toMatchSnapshot();
  });
});
