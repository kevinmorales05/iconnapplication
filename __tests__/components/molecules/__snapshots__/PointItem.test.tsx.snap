// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <PointItem /> should match with the snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  hitSlop={
    Object {
      "bottom": undefined,
      "left": undefined,
      "right": undefined,
      "top": undefined,
    }
  }
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flex": 0,
      "height": undefined,
      "marginBottom": undefined,
      "marginHorizontal": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "marginVertical": undefined,
      "opacity": 1,
      "width": undefined,
      "zIndex": 2,
    }
  }
>
  <View
    pointerEvents="none"
  >
    <View
      collapsable={false}
      style={
        Array [
          Array [
            Object {
              "flexDirection": "column",
            },
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
            Object {},
          ],
          Object {
            "height": 85,
            "paddingHorizontal": 16,
            "paddingTop": 14,
          },
        ]
      }
    >
      <View
        collapsable={false}
        style={
          Array [
            Array [
              Object {
                "flexDirection": "column",
              },
              Object {
                "flexDirection": "row",
              },
              Object {},
              Object {},
              Object {},
              Object {
                "alignItems": "center",
              },
              Object {},
              Object {},
              Object {
                "justifyContent": "space-between",
              },
              Object {},
              Object {
                "height": "60%",
              },
              Object {},
              Object {},
            ],
            undefined,
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Array [
              Array [
                Object {
                  "flexDirection": "column",
                },
                Object {},
                Object {},
                Object {},
                Object {},
                Object {
                  "alignItems": "center",
                },
                Object {},
                Object {},
                Object {},
                Object {
                  "width": "15%",
                },
                Object {},
                Object {},
                Object {},
              ],
              undefined,
            ]
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "testUri": "../../../src/assets/images/iconn-branches-icon-7eleven.png",
              }
            }
            style={
              Object {
                "height": 35,
                "right": 8,
                "top": 0,
                "width": 35,
              }
            }
          />
        </View>
        <View
          collapsable={false}
          style={
            Array [
              Array [
                Object {
                  "flexDirection": "column",
                },
                Object {},
                Object {
                  "flex": 1,
                },
                Object {},
                Object {},
                Object {},
                Object {},
                Object {},
                Object {},
                Object {
                  "width": "60%",
                },
                Object {},
                Object {},
                Object {},
              ],
              Object {
                "left": -8,
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Array [
                Array [
                  Object {
                    "flexDirection": "column",
                  },
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                ],
                Object {
                  "marginBottom": undefined,
                  "marginHorizontal": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "marginVertical": undefined,
                },
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              style={
                Array [
                  Array [
                    Object {
                      "color": "#000000",
                      "fontSize": 15,
                    },
                    Object {
                      "fontSize": 10,
                    },
                  ],
                  Object {
                    "alignSelf": undefined,
                    "letterSpacing": undefined,
                    "lineHeight": undefined,
                    "textAlign": undefined,
                    "textDecorationLine": "none",
                    "textTransform": undefined,
                  },
                ]
              }
            >
              string
            </Text>
          </View>
          <View
            collapsable={false}
            style={
              Array [
                Array [
                  Object {
                    "flexDirection": "column",
                  },
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                ],
                Object {
                  "marginBottom": undefined,
                  "marginHorizontal": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "marginVertical": undefined,
                },
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={4}
              style={
                Array [
                  Array [
                    Object {
                      "color": "#000000",
                      "fontSize": 15,
                    },
                    Object {
                      "fontSize": 10,
                    },
                  ],
                  Object {
                    "alignSelf": undefined,
                    "letterSpacing": undefined,
                    "lineHeight": undefined,
                    "textAlign": undefined,
                    "textDecorationLine": "none",
                    "textTransform": undefined,
                  },
                ]
              }
            >
              string
            </Text>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Array [
              Array [
                Object {
                  "flexDirection": "column",
                },
                Object {
                  "flexDirection": "row",
                },
                Object {},
                Object {
                  "justifyContent": "flex-end",
                },
                Object {},
                Object {},
                Object {},
                Object {},
                Object {},
                Object {
                  "width": "25%",
                },
                Object {},
                Object {},
                Object {},
              ],
              undefined,
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Array [
                Array [
                  Object {
                    "flexDirection": "column",
                  },
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                ],
                Object {
                  "marginBottom": undefined,
                  "marginHorizontal": undefined,
                  "marginLeft": undefined,
                  "marginRight": 20,
                  "marginTop": undefined,
                  "marginVertical": undefined,
                },
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              style={
                Array [
                  Array [
                    Object {
                      "color": "#000000",
                      "fontSize": 14,
                    },
                    Object {
                      "color": "#008060",
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                  ],
                  Object {
                    "alignSelf": undefined,
                    "letterSpacing": undefined,
                    "lineHeight": undefined,
                    "textAlign": undefined,
                    "textDecorationLine": "underline",
                    "textTransform": undefined,
                  },
                ]
              }
            >
              Detalles
            </Text>
          </View>
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#008060",
                  "fontSize": 24,
                },
                Object {
                  "bottom": 4,
                  "right": 8,
                },
                Object {
                  "fontFamily": "Octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          Array [
            Array [
              Object {
                "flexDirection": "column",
              },
              Object {
                "flexDirection": "row",
              },
              Object {},
              Object {
                "justifyContent": "flex-start",
              },
              Object {},
              Object {},
              Object {},
              Object {},
              Object {},
              Object {},
              Object {
                "height": "40%",
              },
              Object {},
              Object {},
            ],
            Object {
              "marginTop": 4,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Array [
              Array [
                Object {
                  "flexDirection": "column",
                },
                Object {},
                Object {},
                Object {},
                Object {},
                Object {},
                Object {},
                Object {},
                Object {},
                Object {
                  "width": "15%",
                },
                Object {},
                Object {},
                Object {},
              ],
              undefined,
            ]
          }
        />
        <View
          collapsable={false}
          style={
            Array [
              Array [
                Object {
                  "flexDirection": "column",
                },
                Object {
                  "flexDirection": "row",
                },
                Object {},
                Object {},
                Object {},
                Object {},
                Object {},
                Object {},
                Object {},
                Object {
                  "width": "70%",
                },
                Object {},
                Object {},
                Object {},
              ],
              Object {
                "left": -8,
              },
            ]
          }
        >
          <View
            collapsable={false}
            style={
              Array [
                Array [
                  Object {
                    "flexDirection": "column",
                  },
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                ],
                Object {
                  "marginBottom": undefined,
                  "marginHorizontal": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "marginVertical": undefined,
                },
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              style={
                Array [
                  Array [
                    Object {
                      "color": "#000000",
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#747476",
                    },
                  ],
                  Object {
                    "alignSelf": undefined,
                    "letterSpacing": undefined,
                    "lineHeight": undefined,
                    "textAlign": undefined,
                    "textDecorationLine": "none",
                    "textTransform": undefined,
                  },
                ]
              }
            >
              DISTANCIA
            </Text>
          </View>
          <View
            collapsable={false}
            style={
              Array [
                Array [
                  Object {
                    "flexDirection": "column",
                  },
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                  Object {},
                ],
                Object {
                  "marginBottom": undefined,
                  "marginHorizontal": undefined,
                  "marginLeft": 8,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "marginVertical": undefined,
                },
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              style={
                Array [
                  Array [
                    Object {
                      "color": "#000000",
                      "fontSize": 12,
                    },
                    Object {
                      "fontWeight": "bold",
                    },
                  ],
                  Object {
                    "alignSelf": undefined,
                    "letterSpacing": undefined,
                    "lineHeight": undefined,
                    "textAlign": undefined,
                    "textDecorationLine": "none",
                    "textTransform": undefined,
                  },
                ]
              }
            >
              string km
            </Text>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            Array [
              Array [
                Object {
                  "flexDirection": "column",
                },
                Object {},
                Object {},
                Object {},
                Object {},
                Object {},
                Object {},
                Object {},
                Object {},
                Object {
                  "width": "15%",
                },
                Object {},
                Object {},
                Object {},
              ],
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
